packages/rich-text/src/component/use-select-object.js

```diff
@@ -9,23 +9,43 @@ export function useSelectObject() {
 			const { target } = event;
 
 			// If the child element has no text content, it must be an object.
-			if ( target === element || target.textContent ) {
+			if (
+				target === element ||
+				( target.textContent && target.isContentEditable )
+			) {
 				return;
 			}
 
 			const { ownerDocument } = target;
 			const { defaultView } = ownerDocument;
-			const range = ownerDocument.createRange();
 			const selection = defaultView.getSelection();
 
+			if ( selection.containsNode( target ) ) return;
+
+			const range = ownerDocument.createRange();
+
 			range.selectNode( target );
 			selection.removeAllRanges();
 			selection.addRange( range );
+
+			event.preventDefault();
+		}
+
+		function onFocusIn( event ) {
+			if (
+				event.relatedTarget &&
+				! element.contains( event.relatedTarget ) &&
+				event.relatedTarget.tagName === 'A'
+			) {
+				onClick( event );
+			}
 		}
 
 		element.addEventListener( 'click', onClick );
+		element.addEventListener( 'focusin', onFocusIn );
 		return () => {
 			element.removeEventListener( 'click', onClick );
+			element.removeEventListener( 'focusin', onFocusIn );
 		};
 	}, [] );
 }
```

Comment:
What are the changes in this file about?

Reply:
Added a comment above